spanner/tests/system/test_system.py

```diff
@@ -728,6 +728,40 @@ def _unit_of_work(transaction, test):
 
         return session, committed
 
+    def test_read_with_multiple_keys_index(self):
+        ROW_COUNT = 10
+        EXTRA_DDL = ['CREATE INDEX name ON contacts(first_name, last_name)']
+        COLUMNS = self.COLUMNS[1], self.COLUMNS[2]
+        pool = BurstyPool()
+
+        temp_db = Config.INSTANCE.database(
+            'test_read_multi_keys_index',
+            ddl_statements=DDL_STATEMENTS + EXTRA_DDL,
+            pool=pool
+        )
+        if not temp_db.exists(): # prevent Conflicts 
+            operation = temp_db.create()
+            self.to_delete.append(_DatabaseDropper(temp_db))
+            operation.result(90)
+        
+        session, committed = self._set_up_table(ROW_COUNT, db=temp_db)
+        self.to_delete.append(session)
+        expected = list(
+            [[row[1], row[2]] for
+            index, row in enumerate(self._row_data(ROW_COUNT))]
+        )
+        keyset = [
+            (expected[index][0], expected[index][1]) for
+            index in range(len(expected))
+        ]
+        rows = list(session.read(self.TABLE,
+                                 COLUMNS,
+                                 KeySet(keys=keyset),
+                                 index='name')
+        )
+        temp_db.drop()
```

Comment:
If we're using a temporary, we need a `try: ... finally: temp_db.drop()` to ensure it gets dropped even if the intermediate code raises.